source/NVDAObjects/window/winword.py

```diff
@@ -1541,23 +1608,59 @@ def script_toggleItalic(self, gesture):
 			# Translators: a message when toggling formatting in Microsoft word
 			ui.message(_("Italic off"))
 
+	@script(gestures=["kb:control+u", "kb:control+shift+u", "kb:control+shift+d"])
 	def script_toggleUnderline(self, gesture):
 		if not self.WinwordSelectionObject:
-			# We cannot fetch the Word object model, so we therefore cannot report the format change.
-			# The object model may be unavailable because this is a pure UIA implementation such as Windows 10 Mail, or its within Windows Defender Application Guard.
-			# Eventually UIA will have its own way of detecting format changes at the cursor. For now, just let the gesture through and don't erport anything.
+			# The object model may be unavailable because this is a pure UIA implementation such as Windows 10 Mail,
+			# or its within Windows Defender Application Guard.
+			# Eventually UIA will have its own way of detecting format changes at the cursor.
+			# For now, just let the gesture through and don't report anything.
 			return gesture.send()
 		val = self._WaitForValueChangeForAction(
 			lambda: gesture.send(),
 			lambda: self.WinwordSelectionObject.font.underline,
 		)
-		if val:
-			# Translators: a message when toggling formatting in Microsoft word
-			ui.message(_("Underline on"))
+		if val != WdUnderline.NONE:
+			try:
+				style = WdUnderline(val).displayString
+				if len(style) > 0:
+					style += " "
+				# Translators: a message when toggling formatting in Microsoft word
+				ui.message(_("Underline{style} on").format(style=style))
```

Comment:
Shouldn't this be the following, the extra space is being appended?

```suggestion
				ui.message(_("Underline {style}on").format(style=style))
```